src/Apps/Order/Routes/Shipping2/AddressModal2.jest.tsx

```diff
@@ -0,0 +1,340 @@
+import {
+  AddressModal,
+  Props,
+  GENERIC_FAIL_MESSAGE,
+  AddressModalActionType,
+} from "Apps/Order/Routes/Shipping2/AddressModal2"
+import { validAddress } from "Components/__tests__/Utils/addressForm"
+import { useSystemContext } from "System/useSystemContext"
+import { SavedAddressType } from "Apps/Order/Routes/Shipping2/Utils/shippingUtils"
+import { createMockEnvironment } from "relay-test-utils"
+import { waitFor } from "@testing-library/react"
+import {
+  ShippingContext,
+  useComputeShippingContext,
+} from "Apps/Order/Routes/Shipping2/Hooks/useShippingContext"
+import { setupTestWrapper } from "DevTools/setupTestWrapper"
+import { graphql } from "react-relay"
+import { AddressModal2TestQuery } from "__generated__/AddressModal2TestQuery.graphql"
+
+jest.unmock("react-relay")
+jest.mock("System/useSystemContext")
+jest.mock("Utils/Hooks/useMatchMedia", () => ({
+  __internal__useMatchMedia: () => ({}),
+}))
+jest.mock("Utils/user", () => ({
+  userHasLabFeature: jest.fn(),
+}))
+
+jest.mock("Apps/Order/Routes/Shipping2/Hooks/useParseOrderData", () => {
+  return {
+    useParseOrderData: () => ({
+      shipsFrom: "US",
+    }),
+  }
+})
+
+const errorBoxQuery = "Banner[data-testid='form-banner-error']"
+
+const savedAddress: SavedAddressType = {
+  ...validAddress,
+  phoneNumber: "8475937743",
+  id: "id",
+  internalID: "internal-id",
+  addressLine3: null,
+  isDefault: false,
+}
+
+let testAddressModalProps: Props
+
+let mockRelayEnv: ReturnType<typeof createMockEnvironment>
+
+beforeEach(() => {
+  mockRelayEnv = createMockEnvironment()
+  ;(useSystemContext as jest.Mock).mockImplementation(() => {
+    return {
+      user: { lab_features: [] },
+      isLoggedIn: true,
+      relayEnvironment: mockRelayEnv,
+    }
+  })
+})
+
+const { getWrapper } = setupTestWrapper<AddressModal2TestQuery>({
+  // Cast props here because we don't actually need the query, just the relay
+  // test wrapper. -
+  // props must be passed via getWrapper's componentProps arg
+  Component: (props: unknown) => {
+    return (
+      <ShippingContext.Provider value={useComputeShippingContext({} as any)}>
+        <AddressModal {...(props as Props)} />
+      </ShippingContext.Provider>
+    )
+  },
+  query: graphql`
+    query AddressModal2TestQuery @relay_test_operation {
+      _unused: artist(id: "whocare") {
+        name
+      }
+    }
+  `,
+})
+
+describe("AddressModal", () => {
+  beforeEach(() => {
+    testAddressModalProps = {
+      onSuccess: jest.fn(),
+      modalAction: {
+        type: AddressModalActionType.EDIT_USER_ADDRESS,
+        address: savedAddress,
+      },
+
+      closeModal: jest.fn(),
+    }
+    ;(useSystemContext as jest.Mock).mockImplementation(() => {
+      return {
+        user: { lab_features: [] },
+        isLoggedIn: true,
+        relayEnvironment: mockRelayEnv,
+      }
```

Comment:
This doesn't need to be duplicated up above in a top-level `beforeEach`, and here in a `describe` `beforeEach`; lets just keep the one inside of the describe block